orchagent/vxlanorch.cpp

```diff
@@ -906,7 +906,11 @@ bool VxlanTunnel::createTunnelHw(uint8_t mapper_list, tunnel_map_use_t map_src,
 
 void VxlanTunnel::deletePendingSIPTunnel()
 {
-   if ((getDipTunnelCnt() == 0) && del_tnl_hw_pending)
+   VxlanTunnelOrch* tunnel_orch = gDirectory.get<VxlanTunnelOrch*>();
+   bool dip_tunnels_used  = tunnel_orch->dipTunnelsUsed();
+
+   if (((dip_tunnels_used && getDipTunnelCnt() == 0) ||
+        !dip_tunnels_used ) && del_tnl_hw_pending)
```

Comment:
Just a suggestion, it looks like we can simplify a bit this expression to avoid using ```dip_tunnels_used``` twice and to do less checks in case of ```dip_tunnels_used == false```.

It could be something like this:
```if ((!dip_tunnels_used || getDipTunnelCnt() == 0) && del_tnl_hw_pending)```

According to "short-circuit evaluation", C++ only evaluates what is necessary from left to right to come up with the result.
* So if ```dip_tunnels_used == false``` then it doesn't check the right side of ```||``` and only checks whether ```del_tnl_hw_pending == true```.
* And only if ```dip_tunnels_used == true``` then it checks if  ```getDipTunnelCnt() == 0``` and only if it is true as well it checks whether ```del_tnl_hw_pending == true```.

Reply:
Thanks for the suggestion. This looks more readable. Updated the code.